tests/functional/u/use/used_before_assignment_issue2615.py

```diff
@@ -4,6 +4,57 @@ def main():
     try:
         res = 1 / 0
         res = 42
+        if main():
+            res = None
+        with open(__file__, encoding="utf-8") as opened_file:
+            res = opened_file.readlines()
     except ZeroDivisionError:
         print(res)  # [used-before-assignment]
     print(res)
+
+
+def nested_except_blocks():
+    """Assignments in an except are tested against possibly failing
+    assignments in try blocks at two different nesting levels."""
+    try:
+        res = 1 / 0
+        res = 42
+        if main():
+            res = None
+        with open(__file__, encoding="utf-8") as opened_file:
+            res = opened_file.readlines()
+    except ZeroDivisionError:
+        try:
+            more_bad_division = 1 / 0
+        except ZeroDivisionError:
+            print(more_bad_division)  # [used-before-assignment]
+            print(res)  # [used-before-assignment]
+    print(res)
```

Comment:
Wise question. The checker still assumes try statements mostly execute--just not when evaluating statements in the except block, since they're written to run when try statements raise. This is outside the except, so it's fine.